sources/engine/Stride.Engine/Engine/Startup/GameBuilder.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+
+namespace Stride.Engine.Startup;
+
+public interface IGameBuilder
+{
+    public ConfigurationManager Configuration { get; }
+    public IServiceCollection Services { get; }
```

Comment:
IServiceCollection standardizes registering dependencies.
It comes with many, many extension methods for convenience.
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.iservicecollection?view=dotnet-plat-ext-7.0#extension-methods

IServiceCollection can not resolve services, however. For that, we need to call
IServiceCollection.BuildServiceProvider() first, to get an IServiceProvider.
This is a different from Stride's ServiceRegistry,
which can have servces added at any time.